core/src/checks/rpc_checks.c

```diff
@@ -0,0 +1,202 @@
+  serialized_request[serialized_size] = 0xaa; // set the last byte to an arbitrary value
+
+  pb_istream_t istream = pb_istream_from_buffer(serialized_request, serialized_size + 1);
+  const size_t response_buffer_size = 2048; // 2048 bytes should be more than enough
+  serialized_response = (uint8_t*) calloc(1, response_buffer_size);
+  if (NULL == serialized_response) {
+    ERROR("%s: calloc(1, %zu) failed", __func__, response_buffer_size);
+    result = -1;
+    goto out;
+  }
+  pb_ostream_t ostream2 = pb_ostream_from_buffer(serialized_response, response_buffer_size);
+  ERROR("(next line is expected to show red text...)");
+
+  // Now that we have a serialized buffer, try to pass it to handle_incoming_message().
+  // This should fail because the InitWallet.random_bytes field has a length of 65 bytes,
+  // but nanopb options set the limit for this field at 64 bytes.
+  //
+  // NOTE: when building for nCipher, there are command hooks that would reject the command
+  // because it's missing the tickets for key use authorization. But this doesn't matter for
+  // this test case, because the protobuf parsing happens before that and fails first.
+  handle_incoming_message(&istream, &ostream2);
+  const size_t actual_response_size = ostream2.bytes_written;
+  if (actual_response_size == 0) {
+    ERROR("%s: no response received from handle_incoming_message(): %s", __func__, PB_GET_ERROR(&ostream2));
+    result = -1;
+    goto out;
+  }
+  pb_istream_t istream2 = pb_istream_from_buffer(serialized_response, actual_response_size);
+  InternalCommandResponse response; // note: no need to init, pb_decode_delimited() does it
+  if (!pb_decode_delimited(&istream2, InternalCommandResponse_fields, &response)) {
+    ERROR(
+        "%s: pb_decode_delimited(..., InternalCommandResponse_fields, ...) failed: %s",
+        __func__,
+        PB_GET_ERROR(&istream2));
+    result = -1;
+    goto out;
+  }
+  if (response.which_response != InternalCommandResponse_Error_tag) {
+    ERROR(
+        "%s: wrong response tag: %d, expected: %d",
+        __func__,
+        (int) response.which_response,
+        (int) InternalCommandResponse_Error_tag);
+    result = -1;
+    goto out;
+  }
+  if (response.response.Error.code != Result_COMMAND_DECODE_FAILED) {
+    ERROR(
+        "%s: wrong response error code: %d, expected: %d",
+        __func__,
+        (int) response.response.Error.code,
+        (int) Result_COMMAND_DECODE_FAILED);
+    result = -1;
+    goto out;
+  }
+  if (!response.response.Error.has_message) {
+    ERROR("%s: error response does not contain a 'message' field", __func__);
+    result = -1;
+    goto out;
+  }
+  if (0 != strcmp("Decode Input failed: bytes overflow", response.response.Error.message)) {
+    ERROR("%s: error response contains unexpected message: %s", __func__, response.response.Error.message);
+    result = -1;
+    goto out;
+  }
+
+out:
+  free(serialized_request);
```

Comment:
Most (all?) of the other tests print an INFO when the tests pass: `verify_bip32 ok`.

```suggestion
  if (result == 0) {
    INFO("rpc_checks: ok");
  }

  free(serialized_request);
```

Reply:
Good idea, will do.